src/Orleans/CodeGeneration/TypeUtils.cs

```diff
@@ -1079,5 +1069,32 @@ private static string EscapeIdentifier(string identifier)
                     return identifier;
             }
         }
+
+        /// <summary>
+        /// Options for formatting type names.
+        /// </summary>
+        public class TypeFormattingOptions
+        {
+            public TypeFormattingOptions(
+                string nameSuffix = null,
+                bool includeNamespace = true,
+                bool includeGenericParameters = true,
+                bool includeTypeParameters = true,
+                char nestedClassSeparator = '.')
+            {
+
+                this.NameSuffix = nameSuffix;
+                this.IncludeNamespace = includeNamespace;
+                this.IncludeGenericTypeParameters = includeGenericParameters;
+                this.IncludeTypeParameters = includeTypeParameters;
+                this.NestedTypeSeparator = nestedClassSeparator;
+            }
+
+            public bool IncludeNamespace { get; set; }
+            public bool IncludeTypeParameters { get; set; }
+            public bool IncludeGenericTypeParameters { get; set; }
+            public char NestedTypeSeparator { get; set; }
+            public string NameSuffix { get; set; }
```

Comment:
Should you be making this whole class immutable (or at least a struct)? Seems too risky to keep passing this object reference around, and somebody might mistakenly mutate it and affect the callers. I'm not saying you are doing that, but I don't know, as it's not trivial to follow if that ever happens (for example you have nullable argument that sometimes forwards the previous options in a recursive call, but sometimes creates a new instance).